betty/extension/webpack/__init__.py

```diff
@@ -0,0 +1,142 @@
+"""
+Integrate Betty with Webpack.
+"""
+from __future__ import annotations
+
+from asyncio import to_thread
+from collections.abc import Callable
+from hashlib import md5
+from json import dumps, loads
+from logging import getLogger
+from pathlib import Path
+from shutil import copytree
+from typing import Any
+
+import aiofiles
+
+from betty.app.extension import Extension
+from betty.asyncio import gather
+from betty.extension.npm import _Npm, npm
+from betty.extension.webpack.jinja2.filter import FILTERS
+from betty.fs import iterfiles
+from betty.generate import Generator, GenerationContext
+from betty.html import CssProvider
+from betty.jinja2 import Jinja2Provider
+
+
+class _WebpackEntrypointProvider:
+    pass
+
+
+class _Webpack(Extension, CssProvider, Jinja2Provider, Generator):
+    @classmethod
+    def depends_on(cls) -> set[type[Extension]]:
+        return {_Npm}
+
+    @classmethod
+    def assets_directory_path(cls) -> Path:
+        return Path(__file__).parent / 'assets'
+
+    @property
+    def public_css_paths(self) -> list[str]:
+        return [
+            self.app.static_url_generator.generate('css/vendor.css'),
+        ]
+
+    def new_context_vars(self) -> dict[str, Any]:
+        return {
+            'webpack_js_entrypoints': set(),
+        }
+
+    @property
+    def filters(self) -> dict[str, Callable[..., Any]]:
+        return FILTERS
+
+    async def generate(self, job_context: GenerationContext) -> None:
```

Comment:
Split this into smaller methods. The actual build code must be callable from the PyInstaller API.